code/modules/mob/living/carbon/xenomorph/abilities/general_abilities.dm

```diff
@@ -479,3 +479,60 @@
 	// Perform check_state(TRUE) silently:
 	if(xeno && !xeno.is_mob_incapacitated() || !xeno.buckled || !xeno.evolving && xeno.plasma_stored >= plasma_cost)
 		return TRUE
+
+/datum/action/xeno_action/onclick/tacmap
+	name = "View Tactical Map"
+	action_icon_state = "toggle_queen_zoom"
+	ability_name = "view tacmap"
+
+	var/mob/living/carbon/xenomorph/queen/tracked_queen
+
+/datum/action/xeno_action/onclick/tacmap/give_to(mob/living/carbon/xenomorph/xeno)
+	. = ..()
+
+	RegisterSignal(xeno.hive, COMSIG_HIVE_NEW_QUEEN, PROC_REF(handle_new_queen))
```

Comment:
is having a hive guaranteed in 100% of all cases?

Reply:
yep!